src/finch/finch_logic/nodes.py

```diff
@@ -170,8 +170,8 @@ class Table(LogicNode):
         idxs: The fields indexing the tensor.
     """
 
-    tns: LogicNode
-    idxs: tuple[LogicNode]
+    tns: Immediate
```

Comment:
Answering your question about missing `fill_value`. If we make this field e.g. `AbstractTensor` from `tensor/tensor.py` and make sure `AbstractTensor` has abstract `fill_value` property, then it should work.